pytorch_lightning/core/lightning.py

```diff
@@ -187,6 +190,18 @@ def automatic_optimization(self) -> bool:
     def automatic_optimization(self, automatic_optimization: bool) -> None:
         self._automatic_optimization = automatic_optimization
 
+    @property
+    def truncated_bptt_steps(self) -> Optional[int]:
+        """
+        truncated_bptt_steps: Truncated back prop breaks performs backprop every k steps of much a longer sequence.
+        If this is > 0, the training step is passed ``hiddens``.
+        """
+        return self._truncated_bptt_steps
```

Comment:
i don't know why the trainer has this as Optional[int]. Could we set this as an int with default = 0?

Reply:
Maybe a noob question, but shouldn't we default it to one? Like from my understanding: If you do one step, you basically perform training without tbtt but if you perform zero steps you don't do anything at all.

Reply:
reading through the code, i think steps might be a misnomer, and the code is indicating splits? where 0 splits means run 1 step, and 1 split = 2 steps, etc? 

https://github.com/PyTorchLightning/pytorch-lightning/blob/e0c64f0ef639a8b9f46e0d8e32e5e0a6b7532cff/pytorch_lightning/core/lightning.py#L1451-L1466

Reply:
@carmocca @awaelchli @Borda what do you think?

Reply:
The only case I can think of where you would want to pass 1 is that maybe somebody wants to get the `hiddens` argument for different reasons without doing any tbptt logic and passing `1` allows that.

But not sure if that makes any sense.

I don't see the point of allowing `None` though, other than differentiating whether the user set this or not.

Reply:
Ok, I will set the default to 0 and make this take int only instead of Optional[int]. This will simplify the validation logic too